internal/ethapi/api.go

```diff
@@ -1332,6 +1332,57 @@ func (s *PublicBlockChainAPI) GetVerifyResult(ctx context.Context, blockNr rpc.B
 	return s.b.Chain().GetVerifyResult(uint64(blockNr), blockHash, diffHash)
 }
 
+// RegisterValidatorArgs has the epoch and the uri to send the proposedBlock to the validator
+type RegisterValidatorArgs struct {
+	Data      hexutil.Bytes `json:"data"` // bytes of string with callback ProposedBlockUri
+	Signature hexutil.Bytes `json:"signature"`
+}
+
+func (s *PublicEthereumAPI) RegisterValidator(ctx context.Context, args RegisterValidatorArgs) error {
+	return nil
+}
+
+type ProposedBlockArgs struct {
+	BlockNumber   rpc.BlockNumber `json:"blockNumber"`
+	PrevBlockHash common.Hash     `json:"prevBlockHash"`
+	BlockReward   *big.Int        `json:"blockReward"`
+	GasLimit      uint64          `json:"gasLimit"`
+	GasUsed       uint64          `json:"gasUsed"`
+	Payload       []hexutil.Bytes `json:"payload"`
+}
+
+// ProposedBlock will submit the block to the miner worker
+func (s *PublicBlockChainAPI) ProposedBlock(ctx context.Context, args ProposedBlockArgs) error {
```

Comment:
Any authorization method to verify the request comes from the relayer?

Reply:
Good, pretend like a blockchain genius but never let it stop.